Assembly-CSharp/ModLoader.cs

```diff
@@ -91,42 +91,67 @@ public static IEnumerator LoadMods(GameObject coroutineHolder)
                 .GetFields(BindingFlags.Public | BindingFlags.Static)
                 .ToDictionary(f => (int)f.GetValue(null), f => f.Name);
 
+            // Called only once for performance
+            string[] modFiles = Directory.GetFiles(path, "*.dll");
+            // List which mods have which other mods as dependencies
+            List<ValueTuple<string, List<string>>> modDeps = new ();
+            // List how deep into the dependency tree a mod is
+            Dictionary<string, int> modDepth = new ();
+            // List of mods per depth level
             Dictionary<int, List<string>> modTypes = new ();
-            foreach (string modPath in Directory.GetFiles(path, "*.dll"))
+
+            foreach (string modPath in modFiles)
             {
+                // 'Register' the mods
                 using AssemblyDefinition asmDef = AssemblyDefinition.ReadAssembly(modPath);
-                CustomAttribute attr = asmDef.CustomAttributes
-                    .FirstOrDefault(a => a.AttributeType.FullName == typeof(AssemblyTypeAttribute).FullName);
+                modDeps.Add((asmDef.MainModule.Name.Replace(".dll", ""), new List<string>()));
+            }
+            foreach (string modPath in modFiles)
+            {
+                // Put the mods' dependencies into the list
+                using AssemblyDefinition asmDef = AssemblyDefinition.ReadAssembly(modPath);
```

Comment:
I'm not familiar with any internal caches, but perhaps caching the `AssemblyDefinition` objects from the previous for loop could ease on the performance too 👍

Reply:
Added